tests/legacy-cli/e2e/tests/misc/update-git-clean-subdirectory.ts

```diff
@@ -0,0 +1,26 @@
+import { createDir, writeFile } from '../../utils/fs';
+import { expectToFail } from '../../utils/utils';
+import { getGlobalVariable } from '../../utils/env';
+import { ng, silentGit } from '../../utils/process';
+import { prepareProjectForE2e } from '../../utils/project'
+
+export default async function() {
+  process.chdir(getGlobalVariable('tmp-root'));
+
+  await createDir('./subdirectory');
+  process.chdir('./subdirectory');
+
+  await silentGit('init', '.');
+
+  await ng('new', 'subdirectory-test-project', '--skip-install');
+  process.chdir('./subdirectory-test-project');
+  await prepareProjectForE2e('subdirectory-test-project');
+
+  await writeFile('../added.ts', 'console.log(\'created\');\n');
+  await silentGit('add', '../added.ts');
+
+  const { message } = await expectToFail(() => ng('update', '--all'));
```

Comment:
@clydin with this use of `expectToFail`, I could use some guidance.

I don't actually expect this to fail, but currently `ng update --all` errors out with warnings (`Incompatible peer dependencies found.`). What alternatives to I have to deal with both error and non-error conditions idiomatically?